unittests/config.cpp

```diff
@@ -0,0 +1,241 @@
+#include <config/Config.h>
+
+#include <boost/log/trivial.hpp>
+#include <boost/log/utility/setup.hpp>
+#include <gtest/gtest.h>
+
+#include <exception>
+#include <iostream>
+
+using namespace clio;
+using namespace boost::log;
+using namespace std;
+namespace json = boost::json;
+
+class ConfigTest : public ::testing::Test
+{
+protected:
+    Config cfg{json::object({
+        {
+            "arr",
+            json::array({
+                json::object({{"first", 1234}}),
+                json::object({{"second", true}}),
+                json::object({{
+                    "inner_section",
+                    json::array{
+                        json::object({{"inner", "works"}}),
+                    },
+                }}),
+                json::array{
+                    "127.0.0.1",
+                    "192.168.0.255",
+                },
+            }),
+        },
+        {
+            "section",
+            json::object({{
+                "test",
+                json::object({
+                    {"str", "hello"},
+                    {"int", 9042},
+                    {"bool", true},
+                }),
+            }}),
+        },
+        {"top", 420},
+    })};
```

Comment:
Why don't we parse an actual config here, such as the example config or some such?

Reply:
It would be nice to pass an actual config but we can't guarantee that the file will be present in the expected location. There are two solutions: don't use a file in tests or specify exactly where the file is expected relative to CWD when running the tests. We can certainly think about setting some expectation. wdyt?

Reply:
Although we can't have a file directly, i am able to make a temporary file, write a JSON string there and use that instead. So we will have it tested as well.